extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -143,6 +162,11 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 			install: installation
 		};
 
+		let existingInstance = this.factory.checkIfInstanceExists(serverInstanceOptions);
```

Comment:
Potential race condition here if multiple things are trying to create a server at the same time. I'd suggest making this into a getOrCreateInstance function instead that handles that correctly so callers don't have to worry about doing the right thing